src/aws/osml/model_runner/tile_worker/tile_worker_utils.py

```diff
@@ -86,46 +90,71 @@ def setup_tile_workers(
 
 def process_tiles(
     tiling_strategy: TilingStrategy,
-    region_request: RegionRequest,
+    region_request_item: RegionRequestItem,
     tile_queue: Queue,
     tile_workers: List[TileWorker],
     raster_dataset: gdal.Dataset,
     sensor_model: Optional[SensorModel] = None,
-) -> Tuple[int, int]:
+) -> tuple[int, int]:
     """
     Loads a GDAL dataset into memory and processes it with a pool of tile workers.
 
     :param tiling_strategy: the approach used to decompose the region into tiles for the ML model
-    :param region_request: RegionRequest = the region request to update.
+    :param region_request_item: RegionRequestItem = the region request to update.
     :param tile_queue: Queue = keeps the image in the queue for processing
-    :param tile_workers: List[Tileworker] = the list of tile workers
+    :param tile_workers: List[TileWorker] = the list of tile workers
     :param raster_dataset: gdal.Dataset = the raster dataset containing the region
     :param sensor_model: Optional[SensorModel] = the sensor model for this raster dataset
 
-    :return: Tuple[int, int] = number of tiles processed, number of tiles with an error
+    :return: tuple[int, int, List[ImageRegion]] = number of tiles processed, number of tiles with an error
     """
 
-    tile_array = tiling_strategy.compute_tiles(
-        region_request.region_bounds, region_request.tile_size, region_request.tile_overlap
+    # Grab completed tiles from region item
+    # Explicitly cast to Tuple[Tuple[int, int], Tuple[int, int]]
+    # Ensure the bounds have exactly two integers before converting
+    region_bounds: Tuple[Tuple[int, int], Tuple[int, int]] = (
+        (region_request_item.region_bounds[0][0], region_request_item.region_bounds[0][1]),
+        (region_request_item.region_bounds[1][0], region_request_item.region_bounds[1][1]),
     )
+
+    # Explicitly cast tile_size to Tuple[int, int]
+    tile_size: Tuple[int, int] = (region_request_item.tile_size[0], region_request_item.tile_size[1])
+
+    # Explicitly cast tile_overlap to Tuple[int, int]
+    tile_overlap: Tuple[int, int] = (region_request_item.tile_overlap[0], region_request_item.tile_overlap[1])
+
+    tile_array = tiling_strategy.compute_tiles(region_bounds, tile_size, tile_overlap)
+
+    if region_request_item.succeeded_tiles is not None:
+        # Filter ImageRegions based on matching in succeeded_tiles
+        filtered_regions = [
+            region
+            for region in tile_array
+            if [[region[0][0], region[0][1]], [region[1][0], region[1][1]]] not in region_request_item.succeeded_tiles
+        ]
+        if len(tile_array) != len(tile_array):
+            logger.warning(f"{len(tile_array) - len(tile_array)} tiles have already been processed!")
```

Comment:
Warning? This seems like a normal occurrence. Suggest logging as info or debug.

Reply:
Fixed